### PR TITLE
Add Continuation Allocation profiling flag

### PR DESCRIPTION
**Pull request by fengxue-IS**
Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

### CODE REVIEW

**Comment by fengxue-IS** [comment]
Sample output format:
```
Total Continuation entries: 5510070

Cache Hits:                 3815157
     T1 Cache Hits:             1034234
     T2 Cache Hits:             2780923
Cache Miss:                 1694913
     Fast Alloc (<10000ns):     1526532
     Avg Fast Alloc Time:       2209ns
     Slow Alloc (>10000ns):     168381
     Avg Slow Alloc Time:       119254ns
Avg Cache Lookup Time:      142ns
```

File: runtime/vm/ContinuationHelpers.cpp [hunk 1]
```diff
@@ -57,14 +63,21 @@ createContinuation(J9VMThread *currentThread, j9object_t continuationObject)
...
 			continuation = NULL;
 		}
 	}
+#if defined(J9VM_PROF_CONTINUATION_ALLOCATION)
+		vm->avgCacheLookupTime += j9time_nano_time() - start;
+#endif /* defined(J9VM_PROF_CONTINUATION_ALLOCATION) */
 
 	/* No cache found, allocate new continuation structure. */
 	if (NULL == continuation) {
+		start = j9time_nano_time();
```
Comment (tajila) [inline runtime/vm/ContinuationHelpers.cpp:80]: should this be ifdef'd

**Review by tajila — APPROVED** [review]
otherwise LGTM

File: runtime/vm/ContinuationHelpers.cpp (hunk 1 above)
Comment (fengxue-IS) [reply runtime/vm/ContinuationHelpers.cpp:81]: fixed

File: runtime/vm/jvminit.c [hunk 2]
```diff
@@ -6296,6 +6296,21 @@ runExitStages(J9JavaVM* vm, J9VMThread* vmThread)
 	}
 #endif
 
+#if defined(J9VM_PROF_CONTINUATION_ALLOCATION)
```
Comment (tajila) [inline runtime/vm/jvminit.c:6299]: do we need JDK21 ifdeefs here as well

Comment (fengxue-IS) [reply runtime/vm/jvminit.c:6299]: `J9VM_PROF_CONTINUATION_ALLOCATION` flag defaults to false, and the enable option is only added for Java20/21, so I assume we don't need ifdef for Java versions

**Comment by tajila** [comment]
jenkins compile win jdk20

**Comment by tajila** [comment]
jenkins test sanity alinux64 jdk20

**Comment by fengxue-IS** [comment]
for reference: sanity.functional https://openj9-jenkins.osuosl.org/job/Pipeline_Build_Test_JDK20_aarch64_linux/202/

**Comment by tajila** [comment]
jenkins compile alinux64 jdk20

File: runtime/vm/jvminit.c [hunk 3]
```diff
@@ -6345,6 +6345,21 @@ runExitStages(J9JavaVM* vm, J9VMThread* vmThread)
...
+		printf("\nCache Hits:                 %u", vm->t1CacheHit + vm->t2CacheHit);
+		printf("\n     T1 Cache Hits:             %u", vm->t1CacheHit);
+		printf("\n     T2 Cache Hits:             %u", vm->t2CacheHit);
+		printf("\nCache Miss:                 %u", vm->fastAlloc + vm->slowAlloc);
+		printf("\n     Fast Alloc (<10000ns):     %u", vm->fastAlloc);
+		printf("\n     Avg Fast Alloc Time:       %ldns", (vm->fastAlloc > 0 ? (vm->fastAllocAvgTime / (I_64)vm->fastAlloc) : 0));
+		printf("\n     Slow Alloc (>10000ns):     %u", vm->slowAlloc);
+		printf("\n     Avg Slow Alloc Time:       %ldns", (vm->slowAlloc > 0 ? (vm->slowAllocAvgTime / (I_64)vm->slowAlloc) : 0));
+		printf("\nAvg Cache Lookup Time:      %ldns\n", (vm->avgCacheLookupTime / (I_64)(vm->t1CacheHit + vm->t2CacheHit + vm->fastAlloc + vm->slowAlloc)));
+	}
```
Comment (keithc-ca) [inline runtime/vm/jvminit.c:6360]: These format strings should be using `OMR_PRIu32`, `OMR_PRIu64`, etc.

**Pull request merged**